### PR TITLE
add debug launcher for vscode

### PR DESCRIPTION
**Pull request by ras1**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (hebron-george) [inline .vscode/launch.json:1]: You should remove the IDE files from git:

```sh
git rm -r .vscode
```

And then add `.vscode*` to `.gitignore`.

File: README.md [hunk 2]
```diff
@@ -5,4 +5,32 @@ suddenly the flow is broken when a video that's been removed or changed country
...
+install postgresql
+
+gem install bundler
+
+bundle install
+
+optional(set env variable for DB_HOST)
+optional(set env variable for DB_PORT)
+set env variable for DB_PASSWORD
+set env variable for DB_USERNAME
```
Comment (hebron-george) [inline README.md:22]:
```suggestion
Create a `.env` file in the root directory of the project (see [dotenv](https://github.com/bkeepers/dotenv) for more details) and set these local env vars:

DB_USERNAME
DB_PASSWORD
DB_HOST (optional)
DB_PORT (optional)
```

File: README.md [hunk 3]
```diff
@@ -5,4 +5,32 @@ suddenly the flow is broken when a video that's been removed or changed country
...
+
+rails db:setup
+
+#Running
```
Comment (hebron-george) [inline README.md:26]:
```suggestion
# Running
```

File: README.md [hunk 4]
```diff
@@ -5,4 +5,32 @@ suddenly the flow is broken when a video that's been removed or changed country
...
+rails s -p 8000
+
+#Debugging
```
Comment (hebron-george) [inline README.md:29]:
```suggestion
# Debugging (VSCode)
```

File: README.md [hunk 5]
```diff
@@ -5,4 +5,32 @@ suddenly the flow is broken when a video that's been removed or changed country
 to get stuck in an infinite loop of trying to load the video.
```
Comment (hebron-george) [inline README.md:5]: You can update this blob lol, I wrote this when I wasn't sober

Comment (hebron-george) [reply README.md:5]: A simple description like:

```
Happy Youtube Watcher, manages tracked Youtube playlists and their histories over time.
```

**Comment by hebron-george** [comment]
Also we have a gem already to drop in breakpoints. If you type `binding.pry` somewhere, it will pause the execution at that line.

File: README.md [hunk 6]
```diff
@@ -5,4 +5,34 @@ suddenly the flow is broken when a video that's been removed or changed country
...
+
+DB_USERNAME
+DB_PASSWORD
+DB_HOST (optional)
+DB_PORT (optional)
+
+rails db:setup
+
+# Running
+rails s -p 8000
```
Comment (hebron-george) [inline README.md:29]: Maybe we should add a note here about the SYT frontend running on port 3000?

File: README.md [hunk 7]
```diff
@@ -1,8 +1,44 @@
...
+The solution? Happy Youtube Watcher.
+
+# Setup
+
+install ruby(2.5.x)
+install postgresql
+
+gem install bundler
+
+bundle install
```
Comment (hebron-george) [inline README.md:20]: I'm not against this, but imo this section isn't really necessary since all ruby/rails projects have this.

Comment (hebron-george) [reply README.md:20]: It's a non-blocker; up to you

File: README.md [hunk 8]
```diff
@@ -1,8 +1,44 @@
...
+
+# Debugging (VSCode)
+
+Run `gem install debase`
+Run `gem install ruby-debug-ide`
+Run the following command to start a debug instance of your code: `rdebug-ide --host 0.0.0.0 --port 1234 --dispatcher-port 1234 -- bin/rails s -p 8000`
+Install `Ruby` extension on vscode(or other debug listener)
+Launch a debug listener session in vscode: `Listen for rdebug-ide`
+Set a breakpoint
+Execute code to hit breakpoint
```
Comment (hebron-george) [inline README.md:44]: You need to add new lines at the end of each of these or make them bullet points or something. It looks like a giant paragraph rn

**Review by hebron-george — APPROVED** [review]

File: README.md (hunk 7 above)
Comment (ras1) [reply README.md]: Leaving it in as a pointer, I might not come back to this for a while and will be nice to have

**Pull request merged**